### PR TITLE
feat(Dropdown): Added container prop to DropdownMenu using React.portal

### PR DESCRIPTION
**Pull request by hsource**
- [ ] Bug fix 
- [X] New feature 
- [ ] Chore 
- [ ] Breaking change 
- [ ] There is an open issue which this change addresses
- [X] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [X] My commits follow the [Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- - [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

This is an improvement of #1513

When a Dropdown is inside a small element with overflow: hidden, it's useful to have the dropdown display directly inside of body or another container. I added the same `container` prop to DropdownMenu. Using a React Portal, this allows the dropdown to have body or any other tag as the parent.

The DropdownMenu passes its own `ref` back to the parent dropdown via an `onMenuRef` callback in the Dropdown context so that clicks outside the DropdownMenu will properly close the menu.

| Without `container` | With `container |
| --- | --- |
| Before container | Testing container |

### CODE REVIEW

**Comment by hsource** [comment]
I just realized this broke some tests! I can take a quick look later today.

**Comment by hsource** [comment]
Fixed the tests! Ready for review again

**Comment by kyletsang** [comment]
@TheSharpieOne, is this something you'd like to add to the API?  If so, I can review + test this.

**Comment by TheSharpieOne** [comment]
@kyletsang yes

File: docs/lib/Components/DropdownsPage.js [hunk 1]
```diff
@@ -406,6 +412,19 @@ DropdownItem.propTypes = {
             {DropdownSetActiveFromChildSource}
           </PrismCode>
         </pre>
+
+        <SectionTitle>container</SectionTitle>
+        <p>
+          Use the <code>container</code> prop to allow the dropdown menu to be placed inside an alternate container through a React Portal. This can be used to allow the dropdown menu to escape a container with the style `overflow: hidden`.
```
Comment (kyletsang) [inline docs/lib/Components/DropdownsPage.js:418]: Let's wrap "overflow: hidden" in code tags.

**Review by kyletsang — CHANGES_REQUESTED** [review]
`getMenuItems` inside `Dropdown` will need to account for the change in the location of the menu so it can query the menu items.  Currently, it won't find anything, so keyboard controls don't work.

**Comment by hsource** [comment]
> `getMenuItems` inside `Dropdown` will need to account for the change in the location of the menu so it can query the menu items. Currently, it won't find anything, so keyboard controls don't work.

Fixed this! Here's experiments with tab, enter, and the down arrow.

Keyboard check

File: src/Dropdown.js [hunk 2]
```diff
@@ -88,14 +97,18 @@ class Dropdown extends React.Component {
...
+
   getMenuCtrl() {
     if (this._$menuCtrl) return this._$menuCtrl;
     this._$menuCtrl = this.getContainer().querySelector('[aria-expanded]');
     return this._$menuCtrl;
   }
 
   getMenuItems() {
-    return [].slice.call(this.getContainer().querySelectorAll('[role="menuitem"]'));
+    return [].slice.call(this.getMenu().querySelectorAll('[role="menuitem"]'));
```
Comment (kyletsang) [inline src/Dropdown.js:111]:
```suggestion
    const menuContainer = this.getMenu() || this.getContainer();
    return [].slice.call(menuContainer.querySelectorAll('[role="menuitem"]'));
```

Tests are failing here, but it's working in the docs site.  Seems like an issue with timing in the test environment.  Normally, the dropdown is supposed to toggle, then render the popper element (which assigns the menu ref), then trigger the focus code. In the tests, this menu ref is not yet assigned...

Anyways, I think we can safely add the old container as a fallback here just to satisfy the tests.  I don't see this being triggered when testing on the docs site.  This seems hacky, but it does follow whatever existing behavior was there previously anyway.

**Comment by kyletsang** [comment]
@hsource, aside from the 1 issue above, this looks pretty good!  Should be good to merge after you make the last change.

File: src/Dropdown.js (hunk 2 above)
Comment (hsource) [reply src/Dropdown.js]: Thanks again for the review! I pushed your fix and added a comment explaining why we do it. I agree it seems hacky, but I don't think it should ever break in real-life usage.

**Review by kyletsang — APPROVED** [review]

**Pull request merged**